knowledge_repo/app/auth_providers/ldap.py

```diff
@@ -0,0 +1,25 @@
+from flask import request, render_template
+from ldap3 import Server, Connection, ALL
+
+from ..models import User
+from ..auth_provider import KnowledgeAuthProvider
+
+
+class LdapAuthProvider(KnowledgeAuthProvider):
+    _registry_keys = ['ldap']
+
+    def init(self):
+        self.server = Server(self.app.config['LDAP_SERVER'], get_info=ALL)
+
+    def prompt(self):
+        return render_template('auth-login-form.html', skip_password=False)
+
+    def validate(self, user):
+        userdn = self.app.config['USERDN_SCHEMA'].format(user.identifier)
```

Comment:
Two small requests:
- Can we change the name of the config key here to `LDAP_USERDN_SCHEMA` (just to make it semantically clear in the config file that these are related)?
- Can we format it using: `.format(user_id=user.identifier)` so that we can use '{user_id}` in the config file instead of `{0}`?